cmd/execution/main.go

```diff
@@ -93,6 +96,22 @@ func main() {
 			flags.IntVar(&syncThreshold, "sync-threshold", 100, "the maximum number of sealed and unexecuted blocks before triggering state syncing")
 			flags.BoolVar(&extensiveLog, "extensive-logging", false, "extensive logging logs tx contents and block headers")
 		}).
+		Module("mutable follower state", func(node *cmd.FlowNodeBuilder) error {
+			// For now, we only support state implementations from package badger.
+			// If we ever support different implementations, the following can be replaced by a type-aware factory
+			state, ok := node.State.(*badgerState.State)
+			if !ok {
+				return fmt.Errorf("only implementations of type badger.State are currenlty supported but read-only state has type %T", node.State)
+			}
+			followerState, err = badgerState.NewFollowerState(
+				state,
+				node.Storage.Index,
+				node.Storage.Payloads,
```

Comment:
seems follower state only need `Index`, `Payloads` and `ProtocolEvents`, why is that?

Reply:
because those are required modules to implement extending chain with headers. If you are curious where is `Blocks`, `Headers`, `Seals` - they are already in `badger.State` which is already created.